client/client-lib/src/query.rs

```diff
@@ -218,8 +273,8 @@ impl<R: Eq + Clone + Debug> QueryStrategy<R> for CurrentConsensus<R> {
     }
 }
 
-pub trait QueryStrategy<R> {
-    fn process(&mut self, response: FedResponse<R>) -> QueryStep<R>;
+pub trait QueryStrategy<IR, OR = IR> {
```

Comment:
@jkitman :+1: / :-1: ? :)

Reply:
Nice, I was thinking we might need something like this eventually.